okhttp/src/main/java/okhttp3/OkHttpClient.java

```diff
@@ -844,24 +846,30 @@ public Builder dispatcher(Dispatcher dispatcher) {
      */
     public Builder protocols(List<Protocol> protocols) {
       // Create a private copy of the list.
-      protocols = new ArrayList<>(protocols);
-
-      // Validate that the list has everything we require and nothing we forbid.
-      if (!protocols.contains(Protocol.HTTP_1_1)) {
-        throw new IllegalArgumentException("protocols doesn't contain http/1.1: " + protocols);
+      final Set<Protocol> uniqueProtocols = new HashSet<>(protocols);
+
+      if (uniqueProtocols.contains(Protocol.H2C) && uniqueProtocols.size() > 1) {
+        // when using h2c prior knowledge, no other protocol should be supported.
+        throw new IllegalArgumentException("protocols containing h2c cannot use other protocols:"
+                + uniqueProtocols);
+      } else if (!uniqueProtocols.contains(Protocol.HTTP_1_1)) {
```

Comment:
bug: this needs to check if uniqueProtocols does not contain H2c

Reply:
This is checked implicitly via the if case above

Reply:
Actually it's not. In the event that someone specifies only H2C, then the second if check evaluates to false, triggering this else if. Caught it in one of the unit tests I was writing.